Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -41,8 +43,9 @@ public final class CRTAWSCredentialsProvider {
                             allocator: Allocator = defaultAllocator) throws {
 
         var staticOptions = aws_credentials_provider_static_options()
-        staticOptions.shutdown_options = CRTAWSCredentialsProvider.setUpShutDownOptions(
-            shutDownOptions: config.shutDownOptions)
+        staticOptions.shutdown_options = ShutdownCallbackCore(config.shutdownCallback)?
```

Comment:
same, if `aws_credentials_provider_new_static()` fails we never do `core.release()`